src/Automation/SnapshotCommand.cs

```diff
@@ -7,29 +7,73 @@
-        public static IReadOnlyCollection<ScanResults> Execute(Config config, IScanTools scanTools)
+        /// <returns>A set of ScanResults objects that describes the result of the command</returns>
+        public static IReadOnlyCollection<ScanResults> ExecuteScanAll(Config config, IScanTools scanTools)
+        {
+            ValidateScanParameters(config, scanTools);
+
+            return GetScanAllResults(config, scanTools, CancellationToken.None);
+        }
+
+        public static Task<ScanResults> ExecuteScanAsync(Config config, IScanTools scanTools, CancellationToken cancellationToken)
+        {
+            ValidateScanParameters(config, scanTools);
+
+            return TriggerScanAsync(config, scanTools, cancellationToken);
+        }
+
+        /// <summary>
+        /// Execute the scan asynchronously
+        /// </summary>
+        /// <param name="config">A set of configuration options</param>
+        /// <param name="scanTools">A set of tools for writing output files,
+        /// creating the expected results format, and finding the target element to scan</param>
+        /// <param name="cancellationToken">A cancellation token through which scanning can be stopped</param>
+        /// <returns>A set of ScanResults objects that describes the result of the command</returns>
+        public static Task<IReadOnlyCollection<ScanResults>> ExecuteScanAllAsync(Config config, IScanTools scanTools, CancellationToken cancellationToken)
+        {
+            ValidateScanParameters(config, scanTools);
+
+            return TriggerScanAllAsync(config, scanTools, cancellationToken);
+        }
+
+        private static void ValidateScanParameters(Config config, IScanTools scanTools)
         {
             if (config == null) throw new ArgumentNullException(nameof(config));
             if (scanTools == null) throw new ArgumentNullException(nameof(scanTools));
             if (scanTools.TargetElementLocator == null) throw new ArgumentException(ErrorMessages.ScanToolsTargetElementLocatorNull, nameof(scanTools));
             if (scanTools.Actions == null) throw new ArgumentException(ErrorMessages.ScanToolsActionsNull, nameof(scanTools));
             if (scanTools.NativeMethods == null) throw new ArgumentException(ErrorMessages.ScanToolsNativeMethodsNull, nameof(scanTools));
+        }
+
+        private static Task<ScanResults> TriggerScanAsync(Config config, IScanTools scanTools, CancellationToken cancellationToken)
+        {
+            return Task.Run<ScanResults>(() => GetScanAllResults(config, scanTools, cancellationToken).FirstOrDefault());
+        }
+
+        private static Task<IReadOnlyCollection<ScanResults>> TriggerScanAllAsync(Config config, IScanTools scanTools, CancellationToken cancellationToken)
+        {
+            return Task.Run<IReadOnlyCollection<ScanResults>>(() => GetScanAllResults(config, scanTools, cancellationToken));
```

Comment:
nit: personal preference but IMO adding `TriggerScanAllAsync` and `TriggerScanAsync` seems unnecessary, since they both only have one line. I would prefer to call `Task.Run` directly in `ExecuteScanAsync` and `ExecuteScanAll`.